frontend/app/common/fetcher.ts

```diff
@@ -73,46 +73,54 @@ const fetcher = methods.reduce<Partial<FetcherObject>>((acc, method) => {
       rurl += (rurl.includes('?') ? '&' : '?') + `site=${siteId}`;
     }
 
-    return fetch(rurl, parameters).then(res => {
-      const date = (res.headers.has('date') && res.headers.get('date')) || '';
-      const timestamp = isNaN(Date.parse(date)) ? 0 : Date.parse(date);
-      const timeDiff = (new Date().getTime() - timestamp) / 1000;
-      StaticStore.serverClientTimeDiff = timeDiff;
+    return fetch(rurl, parameters)
+      .then(res => {
+        const date = (res.headers.has('date') && res.headers.get('date')) || '';
+        const timestamp = isNaN(Date.parse(date)) ? 0 : Date.parse(date);
+        const timeDiff = (new Date().getTime() - timestamp) / 1000;
+        StaticStore.serverClientTimeDiff = timeDiff;
 
-      if (res.status >= 400) {
-        if (httpErrorMap.has(res.status)) {
-          const errString = httpErrorMap.get(res.status)!;
-          throw {
-            code: -1,
-            error: errString,
-            details: errString,
-          };
-        }
-        return res.text().then(text => {
-          let err;
-          try {
-            err = JSON.parse(text);
-          } catch (e) {
-            if (logError) {
-              // eslint-disable-next-line no-console
-              console.error(err);
-            }
+        if (res.status >= 400) {
+          if (httpErrorMap.has(res.status)) {
+            const descriptor = httpErrorMap.get(res.status) || httpMessages.unexpectedError;
             throw {
-              code: -1,
-              error: 'Something went wrong.',
-              details: text,
+              code: descriptor ? res.status : 500,
```

Comment:
`descriptor` should be true always because if you can't get descriptor by status, you always define it as `httpMessages.unexpectedError`

Reply:
good catch